library/std/src/io/stdio.rs

```diff
@@ -1202,6 +1202,59 @@ where
     }
 }
 
+/// Trait to determine if stdio stream (Stdin, Stdout, Stderr) is a terminal/tty.
+#[unstable(feature = "is_terminal", issue = "80937")]
+pub trait IsTerminal {
+    /// returns true if stdio stream (Stdin, Stdout, Stderr) is a terminal/tty.
+    fn is_terminal() -> bool;
```

Comment:
This should maybe return an `Option` of `bool` where `None` will be returned when we are unable to determine if it's either a terminal or not.
```suggestion
    fn is_terminal() -> Option<bool>;
```

Reply:
This could be a solution to the Windows hack problem. Instead of falling back to the `msys_tty_on` hack, we just return None? Although I'm not sure how useful this would be to the end user, as I'm assuming that any `None`s would have to be treated the same as `Some(false)` in the calling code.

Reply:
I think it'd be appropriate to return `io::Result<bool>` here. If we get the *expected* error that indicates "not a terminal" we should return `false`, but if we get an *unexpected* error we should return that error